### PR TITLE
enhancement: Add a new enhancement 'disk-reconnection'

### PR DESCRIPTION
**Pull request by shuo-wu**
#1269

### CODE REVIEW

File: enhancements/20201106-disk-reconnection.md [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+#### Migrate the existing disks to new nodes
+1. Detach all related volumes using the disk before the disk migration.
+2. Directly move the disk to the new node (physically or in cloud vendor) and mount the disk.
+3. Add the disk with the new mount point to the corresponding new Longhorn node in Longhorn Node page.
+4. Attach the volumes for the workloads.
+
+#### Scale down the node but reuse the disks on the node
+1. Directly shut down the node when there are replicas on the node. Then the replicas on the node will fail.
+2. Move the disks on the down node to other running nodes (physically or in cloud vendor).
+3. Click `Connect Disk` with new the mount point and the new node ID in Longhorn Node page.
```
Comment (yasker) [inline enhancements/20201106-disk-reconnection.md:44]: `Connect Disk`?

File: enhancements/20201106-disk-reconnection.md [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+4. Wait then verify the node ID of the failed replicas will be updated.
+5. Longhorn will reuse the failed replicas to speed up the rebuilding once the feature [#1304](https://github.com/longhorn/longhorn/issues/1304) is done.
+
+### API Changes
+Disk related HTTP APIs are required.
```
Comment (yasker) [inline enhancements/20201106-disk-reconnection.md:49]: Need to list the changes here.

**Review by yasker — CHANGES_REQUESTED** [review]
LGTM except a few minor issues.

File: enhancements/20201106-disk-reconnection.md (hunk 2 above)
Comment (shuo-wu) [reply enhancements/20201106-disk-reconnection.md:49]: Sorry for the mistake. This is the leftover of the previous version.

**Review by yasker — APPROVED** [review]

**Pull request merged**